sql/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -208,7 +236,8 @@ private void initCurrentReader(FileInput fileInput, URI uri) throws IOException
         InputStream stream = fileInput.getStream(uri);
         if (stream != null) {
             currentReader = createBufferedReader(stream);
-            currentLineNumber = 0;
+            csvHeader = Optional.ofNullable(isInputCsv() ? currentReader.readLine() : null);
```

Comment:
I would setup a new CsvLineContext here which reads the first line as the header internally. Otherwise a JsonLineContext is constructed.